index.js

```diff
@@ -101,7 +101,7 @@ function asPromise(opts) {
 
 					res.body = data;
 
-					if (opts.json && statusCode !== 204) {
+					if (opts.json && res.body) {
```

Comment:
Wonder if this makes more sense:

```js
if (opts.json && statusCode !== 204 && res.body) {
```

This way we don't try to read the body, even if there is one when the server tells us there shouldn't be one.

@floatdrop Thoughts?